src/control/system/raft.go

```diff
@@ -84,6 +92,151 @@ func (ro raftOp) String() string {
 	}[ro]
 }
 
+// ResignLeadership causes this instance to give up its raft
+// leadership state. No-op if there is only one replica configured.
+func (db *Database) ResignLeadership(cause error) error {
+	if cause == nil {
+		cause = errors.New("unknown error")
+	}
+	db.log.Errorf("resigning leadership (%s)", cause)
+	if err := db.raft.withReadLock(func(svc raftService) error {
+		return svc.LeadershipTransfer().Error()
+	}); err != nil {
+		return errors.Wrap(err, cause.Error())
+	}
+	return cause
+}
+
+// ShutdownRaft signals that the raft implementation should shut down
+// and release any resources it is holding. Blocks until the shutdown
+// is complete.
+func (db *Database) ShutdownRaft() error {
+	db.log.Debug("shutting down raft instance")
+	return db.raft.withReadLock(func(svc raftService) error {
+		return svc.Shutdown().Error()
+	})
+}
+
+// configureRaft sets up the raft service in preparation for starting it.
+func (db *Database) configureRaft() error {
+	if db.raftTransport == nil {
+		return errors.New("no raft transport configured")
+	}
+
+	rc := raft.DefaultConfig()
+	rc.Logger = newHcLogger(db.log)
+	rc.SnapshotThreshold = 16 // arbitrarily low to exercise snapshots
+	//rc.SnapshotInterval = 5 * time.Second
```

Comment:
Intended to be uncommented?

Reply:
Still working on tuning these parameters.